### PR TITLE
Proguard fix

### PR DESCRIPTION
**Pull request by azizkayumov**
Default [ProGuard](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/build.gradle) will obfuscate and alter the name of your library classes, making it impossible for Android projects to reference them. Here are steps to reproduce the issue:
1. Created a [dummy android project](https://github.com/AbduazizKayumov/Dummy-Payme-Android-SDK-test/blob/master/app/src/main/java/com/piapps/paymeandroidsdktest/MainActivity.kt)
2. Added `implementation 'uz.paycom:payment:1.0.5.5'` to `build.gradle`
3. Refreshed and AS downloaded Payme Android SDK.
4. Tried to follow the [README](https://github.com/PaycomUZ/AndroidSDK/blob/master/README.md)
5. Could not reference EXTRA_ID, EXTRA_AMOUNT from [PaymentActivity.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/PaymentActivity.java). 

Can I assume that the [default Proguard settings](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/proguard-payment.pro) with [`minifyEnabled=True`](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/build.gradle) is causing the issue?  
I previewed the library on AS from libs folder, I can see that Payment Android SDK is full of java classes with "a"s and "b"s. 
Screen Shot 2019-12-17 at 3 27 45 PM

In this PR I added some basic "consumer" rules for payment lib. Let me know your thoughts on this. 
A typical library's Proguard rules:
https://stuff.mit.edu/afs/sipb/project/android/sdk/android-sdk-linux/tools/proguard/docs/index.html#manual/examples.html

### CODE REVIEW

File: payment/build.gradle [hunk 1]
```diff
@@ -26,6 +26,7 @@ android {
     release {
       minifyEnabled true
       proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-payment.pro'
+      consumerProguardFiles 'consumer-proguard-payment.pro'
```
Comment (MatnazaroFF) [inline payment/build.gradle]: I highly recommend not to do this

Comment (azizkayumov) [reply payment/build.gradle]: Let me know the reason or reference why not to do this? I could not use the lib as it is documented in README. What would be the recommended steps to include the SDK in my app?

Comment (MatnazaroFF) [reply payment/build.gradle]: You can just add `@Keep `annotation for constants

Comment (azizkayumov) [reply payment/build.gradle]: Yes, I added @keep annotations for all the public and their protected classes as a general rule. Can you review the consumer-proguard-payment.pro from my PR? I found a typical library's proguard rules as mentioned in here:
https://stuff.mit.edu/afs/sipb/project/android/sdk/android-sdk-linux/tools/proguard/docs/index.html#manual/examples.html
The thing is I can simply add @keep for all the classes and constants, but the lib doesn't stop here, maybe, other things will be available for users in future. So a general proguard rule for a typical library from MIT should not hurt anyone.

**Comment by MatnazaroFF** [comment]
Another option for [proguard-payment.pro](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/proguard-payment.pro)
Screenshot from 2019-12-17 12-47-01
Screenshot from 2019-12-17 12-47-17
Screenshot from 2019-12-17 12-47-41

File: payment/build.gradle (hunk 1 above)
Comment (MatnazaroFF) [reply payment/build.gradle]: https://youtu.be/aBm5iYg7uJU?t=1485

Comment (AbduazizKayumov) [reply payment/build.gradle]: I got your point, what about the latest commit? The docs on[ README.md](https://github.com/PaycomUZ/AndroidSDK/blob/master/README.md) says one can use the following classes: 
[PaymentActivity.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/PaymentActivity.java) constants, [PaycomSandBox.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/utils/PaycomSandBox.java) and [Result.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/model/Result.java).
I tried to make it as defensive as possible:

```
-keep class uz.paycom.payment.PaymentActivity {
    public static <fields>;
}

-keep class uz.paycom.payment.utils.PaycomSandBox { *; }
-keep class uz.paycom.payment.model.Result { *; }
```

**Comment by azizkayumov** [comment]
I got your point, what about the latest commit? The docs on[ README.md](https://github.com/PaycomUZ/AndroidSDK/blob/master/README.md) says one can use the following classes: 
[PaymentActivity.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/PaymentActivity.java) constants, [PaycomSandBox.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/utils/PaycomSandBox.java) and [Result.java](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/model/Result.java).
I tried to make it as defensive as possible:

```
-keep class uz.paycom.payment.PaymentActivity {
    public static <fields>;
}

-keep class uz.paycom.payment.utils.PaycomSandBox { *; }
-keep class uz.paycom.payment.model.Result { *; }
```

**Comment by MatnazaroFF** [comment]
@AbduazizKayumov Perfectly. You understood me correctly.

**Comment by YadgarovIslombek** [comment]
?

**Comment by YadgarovIslombek** [comment]
> Standart [ProGuard](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/build.gradle) sizning kutubxona sinflaringiz nomini [buzadi](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/build.gradle) va o'zgartiradi, shuning uchun Android loyihalarida ularga murojaat qilish imkonsiz bo'ladi. Muammoni takrorlash uchun quyidagi qadamlar:
> 
> 1. A yaratilgan [model android loyihani](https://github.com/AbduazizKayumov/Dummy-Payme-Android-SDK-test/blob/master/app/src/main/java/com/piapps/paymeandroidsdktest/MainActivity.kt)
> 2. Yükladi `implementation 'uz.paycom:payment:1.0.5.5'`uchun`build.gradle`
> 3. Yangilangan va AS yuklab olingan Payme Android SDK.
> 4. [README-](https://github.com/PaycomUZ/AndroidSDK/blob/master/README.md) ni kuzatishga harakat qildim[](https://github.com/PaycomUZ/AndroidSDK/blob/master/README.md)
> 5. EXTRA_ID, EXTRA_AMOUNT, PaymentActivity.java-dan ma'lumot [olinmadi](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/src/main/java/uz/paycom/payment/PaymentActivity.java) .
> 
> Men deb taxmin qilish mumkin [Odatiy Proguard parametrlarini](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/proguard-payment.pro) bilan [`minifyEnabled=True`](https://github.com/PaycomUZ/AndroidSDK/blob/master/payment/build.gradle)muammoni hal nima bo'ladi?
> Libs papkasida AS-dagi kutubxonani oldindan ko'rib chiqdim, Payment Android SDK-ning java sinflariga to'la "a" va "b" lar bilan to'ldirilganligini ko'rdim.
> 2019 yil 17-iyun, soat 3 27 45 da ekran tasviri
> 
> Ushbu PR-da lib uchun ba'zi bir "iste'molchilar" qoidalarini qo'shdim. Bu boradagi fikrlaringizni menga xabar bering.
> Odatda kutubxonaning Proguard qoidalari:
> https://stuff.mit.edu/afs/sipb/project/android/sdk/android-sdk-linux/tools/proguard/docs/index.html#manual/examples.html

????????????????

**Comment by Elbek-hub** [comment]
implementation 'uz.paycom:payment:1.0.5.5'

Shuni aniqlamayapti ?

**Pull request closed without merging**